src/onegov/search/indexer.py

```diff
@@ -306,8 +307,105 @@ def delete(self, task):
                     pass
 
 
-class TypeMapping:
+class PostgresIndexer(Indexer):
+    TEXT_SEARCH_COLUMN_NAME = 'fts_idx'
+
+    def __init__(self, mappings, queue, es_client, engine, session):
+        super().__init__(mappings, queue, es_client, hostname=None)
+
+        self.engine = engine
+        self.session = session
+
+        self.idx_language_mapping = {
+            'de': 'german',
+            'fr': 'french',
+            'it': 'italian',
+            'en': 'english',
+        }
+
+    def index(self, tasks) -> bool:
+        """ Update the 'fts_idx' column (full text search index) of the given
+        object(s)/task(s).
+
+        In case of a bunch of tasks we are assuming they are all from the
+        same schema and table in order to optimize the indexing process.
+
+        :param tasks: A list of tasks to index
+        :return: True if the indexing was successful, False otherwise
+        """
+        content = []
+
+        if not isinstance(tasks, list):
+            tasks = [tasks]
+
+        try:
+            for task in tasks:
+                language = (
+                    self.idx_language_mapping.get(task['language'], 'simple'))
+                data = {
+                    k: str(v)
+                    for k, v in task['properties'].items()
+                    if not k.startswith('es_')}
+                _id = str(task['id'])
+                content.append(
+                    {'language': language, 'data': data, '_id': _id})
+
+            schema = tasks[0]['schema']
+            tablename = tasks[0]['tablename']
+            id_key = tasks[0]['id_key']
+            table = sqlalchemy.table(
+                tablename,
+                (id_col := sqlalchemy.column(id_key)),
+                sqlalchemy.column(self.TEXT_SEARCH_COLUMN_NAME),
+                schema=schema  # type: ignore
+            )
+            tsvector_expr = sqlalchemy.text(
+                'to_tsvector(:language, :data)').bindparams(
+                sqlalchemy.bindparam('language', type_=sqlalchemy.String),
+                sqlalchemy.bindparam('data', type_=sqlalchemy.JSON)
+            )
+            stmt = (
+                sqlalchemy.update(table)
+                .where(id_col == sqlalchemy.bindparam('_id'))
+                .values({self.TEXT_SEARCH_COLUMN_NAME: tsvector_expr})
+            )
+            connection = self.engine.connect()
+            trans = connection.begin()
+            connection.execute(stmt, content)  # executemany supported with
```

Comment:
Unfortunately `executemany` is not supported in sqlalchemy 1.3

Reply:
I believe SQLAlchemy will automatically use either `execute` or `executemany` on the underlying DBAPI 2.0 cursor underneath depending on whether you pass a sequence of dictionaries or a single dictionary. So I don't think we need to worry about this.

I don't think they ever matched DBAPI 2.0 and separated it into two functions.